plugins/pencilblue/controllers/actions/admin/sites/delete_site.js

```diff
@@ -30,12 +30,13 @@ module.exports = function DeleteSiteActionModule(pb) {
         var self = this;
         var siteid = self.pathVars.siteid;
         var siteQueryService = new pb.SiteQueryService();
+        var SiteService = new pb.SiteService();
         siteQueryService.getCollections(function(err, allCollections) {
-            siteQueryService.deleteSiteSpecificContent(allCollections, siteid, function(err, result) {
-                if(util.isError(err)) {
+            SiteService.deleteSiteSpecificContent(allCollections, siteid, function(err, result) {
```

Comment:
Ah. I like it. I'll play with this approach and update the PR ASAP. Thanks Brian!